From 4293c1c0705bd3bb431cc19f081a1c5d7ca524c7 Mon Sep 17 00:00:00 2001
From: Antonio Borneo <antonio.borneo@foss.st.com>
Date: Tue, 28 Jan 2025 15:24:16 +0100
Subject: [PATCH 1584/1695] dts: stm32: use EXTI interrupt for RTC nodes

Now EXTI driver is able to work as interrupt controller.
Switch RTC to use the EXTI interrupt.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 8670b4b6d77c61608731280e085a5d4d0100e5ee dts: stm32: use EXTI interrupt for RTC nodes]
Signed-off-by: Antonio Borneo <antonio.borneo@foss.st.com>
Change-Id: I737d5957149242b22fe1872acc36c3dda55ebc28
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/434930
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/arch/arm/dts/stm32mp211.dtsi | 2 +-
 core/arch/arm/dts/stm32mp231.dtsi | 2 +-
 core/arch/arm/dts/stm32mp251.dtsi | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/core/arch/arm/dts/stm32mp211.dtsi b/core/arch/arm/dts/stm32mp211.dtsi
index 3630073f9..42467845c 100644
--- a/core/arch/arm/dts/stm32mp211.dtsi
+++ b/core/arch/arm/dts/stm32mp211.dtsi
@@ -1079,7 +1079,7 @@
 			reg = <0x46000000 0x400>;
 			clocks = <&rcc CK_BUS_RTC>, <&rcc RTC_CK>;
 			clock-names = "pclk", "rtc_ck";
-			interrupts = <GIC_SPI 13 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 22 IRQ_TYPE_EDGE_RISING>;
 		};
 
 		tamp: tamp@46010000 {
diff --git a/core/arch/arm/dts/stm32mp231.dtsi b/core/arch/arm/dts/stm32mp231.dtsi
index 7b38f5713..5aedfd78e 100644
--- a/core/arch/arm/dts/stm32mp231.dtsi
+++ b/core/arch/arm/dts/stm32mp231.dtsi
@@ -1131,7 +1131,7 @@
 			clock-names = "pclk", "rtc_ck";
 			wakeup-source;
 			wakeup-parent = <&exti2 0xffff 0>;
-			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 22 IRQ_TYPE_EDGE_RISING>;
 		};
 
 		tamp: tamp@46010000 {
diff --git a/core/arch/arm/dts/stm32mp251.dtsi b/core/arch/arm/dts/stm32mp251.dtsi
index 35242bf24..d6b337228 100644
--- a/core/arch/arm/dts/stm32mp251.dtsi
+++ b/core/arch/arm/dts/stm32mp251.dtsi
@@ -1218,7 +1218,7 @@
 			clock-names = "pclk", "rtc_ck";
 			wakeup-source;
 			wakeup-parent = <&exti2 0xffff 0>;
-			interrupts = <GIC_SPI 15 IRQ_TYPE_LEVEL_HIGH>;
+			interrupts-extended = <&exti2 22 IRQ_TYPE_EDGE_RISING>;
 		};
 
 		tamp: tamp@46010000 {
-- 
2.47.2

